src/origin_sdk/gql/queries/input_queries/technology.py

```diff
@@ -54,62 +55,74 @@
 
 
 def get_endo_param_and_def_tree(
-    endo_params: Optional[List[str]] = None, endo_defs: Optional[List[str]] = None
-):
-    return {
-        **(
-            {
-                "parameters": {
-                    param: yearly_values_with_transform for param in endo_params
-                }
-            }
-            if endo_params is not None
-            else {}
-        ),
-        **(
-            {
-                "definitions": {
-                    param: variable_values_with_transform for param in endo_defs
-                }
-            }
-            if endo_defs is not None
-            else {}
-        ),
-    }
+    endo_params_yearly: Optional[List[str]] = None,
+    endo_params_monthly: Optional[List[str]] = None,
+    endo_defs: Optional[List[str]] = None,
+) -> dict:
+    result = {}
+
+    # Handling endo_params_yearly
+    if endo_params_yearly:
+        result["parameters"] = {
+            param: yearly_values_with_transform for param in endo_params_yearly
+        }
+    # Handling endo_params_monthly
+    if endo_params_monthly:
+        result.setdefault("parameters", {}).update(
+            {param: monthly_values_with_transform for param in endo_params_monthly}
+        )
+    # Handling endo_defs
+    if endo_defs:
+        result["definitions"] = {
+            param: variable_values_with_transform for param in endo_defs
+        }
+
+    return result
```

Comment:
This is the pattern I really dislike where we early init and then conditionally mutate the variable.

Not only this, but you're refactoring the entire thing instead of amending it to suit your new use case. Why? You're changing 3x the lines of code for a subjective change?